pages/tasks/[taskId].tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { NextPage, GetServerSideProps } from 'next';
+import fetch from 'node-fetch';
+import Error from 'next/error';
+import { Task } from '../../models/task/task.types';
+import Layout from '../../components/shared/layout/Layout';
+import Header from '../../components/header/Header';
+import { API_URL } from '../../lib/consts/apiUrl';
+
+interface TaskProps {
+  task?: Task;
+  errorCode?: number;
+}
+
+const TaskDetails: NextPage<TaskProps> = ({ errorCode, task }) => {
+  return (
+    <Layout title="Home page">
+      <Header />
+      {task && task.title}
+      {errorCode && (
+        <div className="-my-16">
+          <Error statusCode={errorCode} />
+        </div>
+      )}
+    </Layout>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<TaskProps> = async ({ params }) => {
+  const res = await fetch(`${API_URL}/api/tasks/${params && params.taskId}`);
```

Comment:
może wyciągnij to z paramsów od razu w argumencie funkcji
`({ params: { taskId } })`